Kernel/Tasks/Process.h

```diff
@@ -605,6 +614,22 @@ class Process final
     ErrorOr<void> validate_mmap_prot(int prot, bool map_stack, bool map_anonymous, Memory::Region const* region = nullptr) const;
     ErrorOr<void> validate_inode_mmap_prot(int prot, bool description_readable, bool description_writable, bool map_shared) const;
 
+    template<size_t Size>
+    static ErrorOr<void> get_syscall_string_argument_into_static_char_buffer(Syscall::StringArgument const& argument, FixedCharBuffer<Size>& buffer)
```

Comment:
Again, no requirement that the buffer have static storage duration. And if we know the size, why do we require the caller to pass it in as an argument? We have a nice return slot we could use for that.